### PR TITLE
Implemented vanilla pack

### LINKED ISSUE #5: Implement vanilla pack

### PR DESCRIPTION
**Pull request by ahuca**
close #5 .

### CODE REVIEW

File: common.ps1
Comment (Colho) [inline common.ps1:1]: Would it be possible to add a small comment at the start of file on what the file does? Starting to get crowded with .ps1 files

Comment (ahuca) [reply common.ps1:1]: For which files do you mean? ps1 files?

Comment (ahuca) [reply common.ps1:1]: Damn you're right GitHub review is quite confusing. Completely missed the small little `common.ps1`

image

File: src/TwinGet.Cli/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "profiles": {
+    "TwinGet.Cli": {
+        "commandName": "Project",
+        "commandLineArgs": "pack C:\\Users\\nhgia\\projects\\twinget\\test\\Test.Utils\\TestTwincatProject\\TestTwincatProject1\\TestPlcProject1\\TestPlcProject1.plcproj"
```
Comment (Colho) [inline src/TwinGet.Cli/Properties/launchSettings.json]: Is it possible to change this to a relative path?

File: src/TwinGet.Core/Packaging/OtherStrings.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿// This file is licensed to you under MIT license.
+
+namespace TwinGet.Core.Packaging;
+
+public static class OtherStrings
```
Comment (Colho) [inline src/TwinGet.Core/Packaging/OtherStrings.cs:5]: Marking this as something that could be renamed in the future, NotificationStrings? StateNotificationStrings? StateStrings?

File: src/TwinGet.TwincatInterface/Exceptions/CouldNotGetSystemManager.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// This file is licensed to you under MIT license.
+
+namespace TwinGet.TwincatInterface.Exceptions;
+
+[Serializable]
+public class CouldNotGetSystemManager : Exception
```
Comment (Colho) [inline src/TwinGet.TwincatInterface/Exceptions/CouldNotGetSystemManager.cs:6]: Could ask the Exception suffix to the class name

File: src/TwinGet.TwincatInterface/Exceptions/CouldNotLookUpPlcProject.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// This file is licensed to you under MIT license.
+
+namespace TwinGet.TwincatInterface.Exceptions;
+
+[Serializable]
+public class CouldNotLookUpPlcProject : Exception
```
Comment (Colho) [inline src/TwinGet.TwincatInterface/Exceptions/CouldNotLookUpPlcProject.cs:6]: Same here, add the Exception suffix

File: src/TwinGet.TwincatInterface/Exceptions/InvalidProjectFileFormat.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
+﻿// This file is licensed to you under MIT license.
+
+namespace TwinGet.TwincatInterface.Exceptions;
+
+[Serializable]
+public class InvalidProjectFileFormat : FormatException
```
Comment (Colho) [inline src/TwinGet.TwincatInterface/Exceptions/InvalidProjectFileFormat.cs:6]: Exception Suffix

**Comment by Colho** [comment]
There were a couple of Exceptions without the Exception suffix, decided its better to make a one comment about it rather than ping it on each file separately.

File: test/Test.Utils/CommandRunnerResult.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    public int ExitCode { get; set; }
+
+    public string Output { get; set; }
+
+    public string Errors { get; set; }
+
+    public bool Success => ExitCode == 0;
+
+    public string AllOuput => Output + Environment.NewLine + Errors;
```
Comment (Colho) [inline test/Test.Utils/CommandRunnerResult.cs]: AllOutput?

File: src/TwinGet.TwincatInterface/Exceptions/InvalidProjectFileFormat.cs (hunk 5 above)
Comment (ahuca) [reply src/TwinGet.TwincatInterface/Exceptions/InvalidProjectFileFormat.cs:6]: foo.

File: test/Test.Utils/CommandRunnerResult.cs (hunk 6 above)
Comment (ahuca) [reply test/Test.Utils/CommandRunnerResult.cs]: Resolved in d4be1a5.

File: src/TwinGet.Core/Packaging/OtherStrings.cs (hunk 2 above)
Comment (ahuca) [reply src/TwinGet.Core/Packaging/OtherStrings.cs:5]: Yeah let's change the name to something better as we progress in the future. I have a feeling things are still changing so didn't bother giving it a better name.

File: src/TwinGet.TwincatInterface/TwincatProject.cs [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        TcSmProjectData? tcSmProject =
+            TwincatUtils.DeserializeXmlFileToProjectData<TcSmProjectData>(filePath);
+
+        string? rootDir = Path.GetDirectoryName(filePath);
+
+        if (tcSmProject.Project.Plc?.Projects is not null)
+        {
+#pragma warning disable CS8604 // Possible null reference argument.
+            plcProjects.AddRange(
+                from ProjectElement project in tcSmProject.Project.Plc.Projects
```
Comment (Colho) [inline src/TwinGet.TwincatInterface/TwincatProject.cs:63]: Feel like this could be LINQed somewhere in the future

**Review by Colho — APPROVED** [review]

**Pull request merged**